### PR TITLE
PKI testing

### LINKED ISSUE #616: Add test to verify TLS certificate chain
Add tests to CI to verify TLS certificate chain is valid:
- hosts without TLS are not able to establish a TLS connection when one is required
- Verify common name of host matches expected destination

### PR DESCRIPTION
**Pull request by itrushkin**
This PR is adding tests to CI to verify TLS certificate chain is valid:

- [x] hosts without TLS are not able to establish a TLS connection when one is required
- [x] Verify common name of host matches expected destination

I have pinned the `step-ca` version to 1.16.3. The latest version does not work.

Closes #616.

### CODE REVIEW

**Review by psfoley — APPROVED** [review]
Looks great, Ilya. Approved!

**Review by igor-davidyuk — APPROVED** [review]
Well done!

File: tests/github/pki_wrong_cn.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        '--template', 'keras_cnn_mnist'
+    ])
+    os.chdir(prefix)
+    subprocess.check_call(['fx', 'workspace', 'certify'])
+
+    # invalid common name
+    test_aggregator_common_name('not_an_fqdn', should_fail=True)
+
+    # case when CA is located on separate machine
+    test_aggregator_common_name('some.fqdn.com', should_fail=False)
```
Comment (itrushkin) [inline tests/github/pki_wrong_cn.py]: I am not sure if during certificate request generation we should allow FQDNs that don't coincide with the FQDN of the current machine. For now, we do it.

**Review by aleksandr-mokrov — APPROVED** [review]

File: tests/github/pki_wrong_cn.py (hunk 1 above)
Comment (aleksandr-mokrov) [reply tests/github/pki_wrong_cn.py]: Current machine can have several FQDNs. I believe it can be discussed outside the scope of this PR.

**Comment by itrushkin** [comment]
Please do not merge this yet. I need to check that gRPC will not accept secure connections with an invalid common name.

**Comment by itrushkin** [comment]
gRPC internally accepts connections with an invalid common name. We have a collaborator name-checking function on the server side to prevent unauthorized connections. I have added the test that expects gRPC client execution to fail with an authentication error for the case when the collaborator certificate has an invalid common name.

**Pull request merged**